stellar-multitenant/pkg/internal/provisioning/manager_test.go

```diff
@@ -30,126 +36,154 @@ func Test_Manager_ProvisionNewTenant(t *testing.T) {
+			NetworkPassphrase:      networkPassphrase,
+			DistributionPrivateKey: distAcc.Seed(),
+		})
 		require.NoError(t, err)
-		defer tenantSchemaConnectionPool.Close()
-
-		expectedTablesAfterMigrationsApplied := []string{
-			"assets",
-			"auth_migrations",
-			"auth_user_mfa_codes",
-			"auth_user_password_reset",
-			"auth_users",
-			"countries",
-			"disbursements",
-			"sdp_migrations",
-			"messages",
-			"organizations",
-			"payments",
-			"receiver_verifications",
-			"receiver_wallets",
-			"receivers",
-			"wallets",
-			"wallets_assets",
+		if sigClientType == signing.DistributionAccountDBSignatureClientType {
+			distAccSigClient, err = signing.NewSignatureClient(signing.DistributionAccountDBSignatureClientType, signing.SignatureClientOptions{
+				NetworkPassphrase:           networkPassphrase,
+				DistAccEncryptionPassphrase: keypair.MustRandom().Seed(),
+				DBConnectionPool:            dbConnectionPool,
+			})
+			require.NoError(t, err)
 		}
-		tenant.TenantSchemaMatchTablesFixture(t, ctx, tenantSchemaConnectionPool, schemaName, expectedTablesAfterMigrationsApplied)
 
-		tenant.AssertRegisteredAssetsFixture(t, ctx, tenantSchemaConnectionPool, []string{"USDC:GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5", "XLM:"})
-		tenant.AssertRegisteredWalletsFixture(t, ctx, tenantSchemaConnectionPool, []string{"Demo Wallet", "Vibrant Assist"})
-		tenant.AssertRegisteredUserFixture(t, ctx, tenantSchemaConnectionPool, userFirstName, userLastName, userEmail)
-	})
+		if sigClientType == signing.DistributionAccountEnvSignatureClientType {
+			assert.IsType(t, &signing.DistributionAccountEnvSignatureClient{}, distAccSigClient)
+		} else {
+			assert.IsType(t, &signing.DistributionAccountDBSignatureClient{}, distAccSigClient)
+		}
 
-	t.Run("provision a new tenant for the pubnet", func(t *testing.T) {
-		tenantName := "myorg-us"
-		userFirstName := "First"
-		userLastName := "Last"
-		userEmail := "email@email.com"
-		organizationName := "My Org"
-		uiBaseURL := "http://localhost:3000"
+		p := NewManager(
+			WithDatabase(dbConnectionPool),
+			WithMessengerClient(&messengerClientMock),
+			WithTenantManager(tenantManager),
+			WithDistributionAccountSignatureClient(distAccSigClient),
+		)
 
-		tnt, err := p.ProvisionNewTenant(ctx, tenantName, userFirstName, userLastName, userEmail, organizationName, uiBaseURL, string(utils.PubnetNetworkType))
+		uiBaseURL := "http://localhost:3000"
+		tnt, err := p.ProvisionNewTenant(ctx, tenantName, user.FirstName, user.LastName, user.Email, user.OrgName, uiBaseURL, string(networkType))
 		require.NoError(t, err)
 
-		schemaName := fmt.Sprintf("sdp_%s", tenantName)
 		assert.Equal(t, tenantName, tnt.Name)
 		assert.Equal(t, uiBaseURL, *tnt.SDPUIBaseURL)
+		if sigClientType == signing.DistributionAccountEnvSignatureClientType {
+			assert.Equal(t, distAcc.Address(), *tnt.DistributionAccount)
+		} else {
+			assert.NotNil(t, *tnt.DistributionAccount)
```

Comment:
💡 You can use `strkey.IsValidEd25519PublicKey` to make sure this is a Stellar public key:
```suggestion
			assert.True(t, strkey.IsValidEd25519PublicKey(*tnt.DistributionAccount))
```